azurelinuxagent/ga/exthandlers.py

```diff
@@ -276,6 +288,73 @@ def run(self):
                       message=detailed_msg)
             return
 
+    def get_artifacts_profile(self):
+        try:
+            self.artifacts_profile = self.protocol.get_artifacts_profile()
+        except Exception as e:
+            msg = u"Exception retrieving VmArtifactsProfile blob: {0}".format(ustr(e))
+            detailed_msg = '{0} {1}'.format(msg, traceback.extract_tb(get_traceback(e)))
+            logger.warn(msg)
+
+            add_event(AGENT_NAME,
+                      version=CURRENT_VERSION,
+                      op=WALAEventOperation.ExtensionProcessing,
+                      is_success=False,
+                      message=detailed_msg)
+
+    def process_goal_state(self, instantiation, sequence_number):
+        fabric_goal_state_changed, fast_track_changed = self.determine_what_changed(instantiation, sequence_number)
+        if fabric_goal_state_changed:
+            self.protocol.set_fast_track(False)
```

Comment:
Isn't sequence number per extension? Then why is `sequence_number_file_path = os.path.join(conf.get_lib_dir(), SEQUENCE_NUMBER_FILE_NAME)` at lib level, and not inside extension level (in extension folders or some other place.)